### PR TITLE
#20899 startup task for postgres to update timestamp fields

### LINKED ISSUE #20899: Spike: saving timezone information
There are places in our system where we save dates formatted for timezones into db columns that do not store the timezone.

1. Store date/times with timezone in database

2. Manipulate them in java/apis/factories, always in UTC

3. When returning from REST/GraphQL, we need to return the date/times in the system offset that is in the company and include the timezone in the returned value

4. Front end should always send timezone.  If no timezone is passed, it will be assumed to be UTC

Places to make sure we are storing dates and using dates appropriately 

1. publishing_bundle, publishing_queue
1.5 publishing_pushed_assets.push_date
2. contentlet date fields
3. contentlet.mod_date, container.mod_date, template.mod_date
4. All the *version_infos (version_ts)
5. push_now
6. workflow_task mod_date/creation_date

### PR DESCRIPTION
**Pull request by nollymar**

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezonesTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class Task210901UpdateDateTimezonesTest {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+        // Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Test
+    public void test() throws Exception {
```
Comment (freddyucv) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezonesTest.java:17]: doc?

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/WebAPI.java [hunk 2]
```diff
@@ -168,14 +168,14 @@ public String dateToHTML(Date myDate) {
 	}
 
 	public String dateToHTMLDateTimeRange(Date from, Date to) {
-		return UtilMethods.dateToHTMLDateTimeRange(from, to, GregorianCalendar.getInstance().getTimeZone());
+		return UtilMethods.dateToHTMLDateTimeRange(from, to, APILocator.getCompanyAPI().getDefaultCompany().getTimeZone());
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/WebAPI.java:171]: maybe this code 

<code>
APILocator.getCompanyAPI().getDefaultCompany().getTimeZone()
</code>

should go into a method in a Util class, something like: TimeZoneUtil.getCurrentTimZone()

**Review by freddyucv — CHANGES_REQUESTED** [review]
are you sure that we have all the cases cover with test here?

File: dotCMS/src/main/java/com/dotmarketing/business/UserFactoryImpl.java [hunk 3]
```diff
@@ -75,7 +75,7 @@ private User getDefaultUser(Company company) {
         defaultUser.setEmailAddress(User.DEFAULT + "@" + company.getMx());
 
         defaultUser.setLanguageId(null);
-        defaultUser.setTimeZoneId(null);
+        defaultUser.setTimeZoneId("UTC");
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/business/UserFactoryImpl.java:78]: this should be a constant somewhere

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+
+
+    boolean updateTable(final String tableName) throws SQLException {
+
+        try (Connection conn = DbConnectionFactory.getConnection()) {
+            final ResultSet results = new DotDatabaseMetaData().getColumnsMetaData(conn, tableName);
+            while (results.next()) {
+                if ("timestamp".equals(results.getString("TYPE_NAME"))) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java:55]: shouldn't be case insensitive

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java:55]: shoudn't be handle as a transaction?

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java:55]: I am not pretty sure about the connection handling
I know a transaction is opening on an upper layer, but the try will close this one and ask for a new one, isn't?

**Review by jdotcms — APPROVED** [review]

**Comment by nollymar** [comment]
> are you sure that we have all the cases cover with test here?

The only new logic added was an upgrade task that has a test. Besides that, all the changes included:
- A modification in  a .sql to declare date fields as timestamps with timezones.
- Replaced all calls where a time zone were obtained to use the company's time zone instead (most of them in jsps files)

Do you see any specific place that should be covered by a test?

File: dotCMS/src/main/java/com/dotmarketing/business/UserFactoryImpl.java (hunk 3 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/business/UserFactoryImpl.java]: Done

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezonesTest.java (hunk 1 above)
Comment (nollymar) [reply dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezonesTest.java:23]: Done

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java (hunk 4 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task210901UpdateDateTimezones.java]: > shouldn't be case insensitive

Done

**Review by freddyucv — APPROVED** [review]

**Pull request merged**